lib/whois/record/parser/whois.corporatedomains.com.rb

```diff
@@ -23,8 +23,29 @@ class WhoisCorporatedomainsCom < BaseIcannCompliant
         self.scanner = Scanners::BaseIcannCompliant, {
             pattern_available: /^No match for/
         }
-      end
 
+        property_supported :domain_id do
+          node('Domain ID')
+        end
+
+        property_supported :expires_on do
+          node('Registry Expiry Date') do |value|
+            parse_time(value)
+          end
+        end
+
+        property_supported :registrar do
+          return unless node('Sponsoring Registrar')
```

Comment:
Hi @weppos - After a bit of digging ... if I'm reading correctly, it seems that all 3 properties are actually included in both responses.  Sponsoring Registrar for example is in both verisign (of course) and in the CD response.  Parser class picks the first part parser (CD) because the ICANN base class supports the prop, but the result is nil as it is matching on "Registrar IANA ID" rather than the CD response's "Sponsoring Registrar IANA ID".  Choices I could think of are to "encourage" the parser_select to NOT select the first parser if the match produces a nil ... but that seems not quite right?  Otherwise, only way I can see to implement is similar to what I sent above.  Am I reading correctly?  If so, how would you like to handle case where both parts include the info?  (also ... of course you're right about the test case sample data ... I included both parts in the CD bit ... easy to fix ... sorry about that).